### PR TITLE
fixes issue with setting defcon

### PR DESCRIPTION
**Pull request by Soulofdestiny**
add test for defcon-settings

### CODE REVIEW

File: t/01openqa.t [hunk 1]
```diff
@@ -3,9 +3,17 @@ use Test::Bot::BasicBot::Pluggable;
...
 
+# basic tests
 ok(my $ob = $bot->load('OpenQA'), 'load openqa module');
 is($bot->tell_direct('perl'), 'I hear Ruby is better than perl..', 'simple messages work');
 is($bot->tell_indirect('!perl'), 'I hear Ruby is better than perl..', 'exclamation mark tags are accepted');
 isnt($bot->tell_indirect('perl'), 'I hear Ruby is better than perl..', 'only responds if addressed');
 
+# defcon tests
+$bot->{handlers}->{openqa}->set(defcon1_allowed => 'foo|bar');
+like($bot->tell_direct('defcon 1'), qr/What did I tell you.*/, 'don\'t allow setting defcon works!');
```
Comment (okurz) [inline t/01openqa.t]: just keep it simple and write "do not" without escaping :-)
And the exclamation mark should not be necessary. It is not like we should be _surprised_ by it actualy working!!! ;-)

File: t/01openqa.t [hunk 2]
```diff
@@ -3,9 +3,17 @@ use Test::Bot::BasicBot::Pluggable;
...
+$bot->{handlers}->{openqa}->set(defcon1_allowed => 'test_user|foo|bar');
+like($bot->tell_direct('defcon 1'), qr/Heads up.*DEFCON increase/, 'set Defcon works!');
```
Comment (okurz) [inline t/01openqa.t]: please use consistent way of writing defcon, i.e. either DEFCON or defcon

**Comment by okurz** [comment]
Please adhere to http://chris.beams.io/posts/git-commit/ , i.e. better write commit message as something like this

```
Fix defcon 1 permissions not read correctly from config

defcon 1 could not be set since f1ca1334 introducing the
staging support as the readout of allowed users from
config was actually not happening as wrongly the 'get'
call was surrounded by '//'.

Adds test for defcon settings.
```

Also see the tiny inline suggestions. If this is too much hassle just tell me and I will change this myself :-)

**Comment by Soulofdestiny** [comment]
merged into one single commit

**Pull request merged**